libs/python/helperApiTest.py

```diff
@@ -0,0 +1,178 @@
+    try:
+        response = requests.post(url, headers=headers, data=payload)
+        data = response.json()
+        response.raise_for_status()
+        if(response.status_code == 200):
+            if data.get("access_token", False):
+                access_token = data['access_token']
+                return access_token
+            else:
+                access_token = data['accessToken']
+                return access_token
+    except requests.exceptions.HTTPError as errh:
+        logging.exception(errh)
+    except requests.exceptions.RequestException as err:
+        logging.exception(err)
+
+def get_api_req(uri, access_token):
+    if not uri.startswith('https://'):
+        uri = 'https://' + uri
+    logging.info(f"The API request URL: {uri}")
+    headers = {
+        'Authorization': 'Bearer ' + access_token,
+    }
+    try:
+        response = requests.get(uri, headers=headers)
+        # response.json()
+        response.raise_for_status()
+        if response.status_code in [200, 201, 204]:
+            logging.info("GET request API call is successful!")
+    except requests.exceptions.HTTPError as errh:
+        logging.error("Failed the API request call")
+        logging.exception(errh)
+    except requests.exceptions.RequestException as err:
+        logging.exception(err)
+
+def get_api_req_basic(uri, username, password):
+    if not uri.startswith('https://'):
+        uri = 'https://' + uri
+    logging.info(f"The API request URL: {uri}")
+    try:
+        response = requests.get(uri, auth=(username, password))
+    # response.json()
+        response.raise_for_status()
+        if response.status_code in [200, 201, 204]:
+            logging.info("GET request API call is successful!")
+    except requests.exceptions.HTTPError as errh:
+        logging.exception(errh)
+        logging.error("Failed the API request call")
+    except requests.exceptions.RequestException as err:
+        logging.exception(err)
+
+def get_key_values(binding, pathKey):
+    for key in pathKey.split("."):
+        if key.isnumeric():
+            key = int(key)
+            binding = binding[key]
+            continue
+        binding = binding.get(key, False)
+        if not binding:
+            break
+    return binding
+
+def main():
+    params = read_json_file('/home/user/usecases/inDevelopment/cross_consumption.json')
+    # print(params)
+    data = params["api_resource_uri"]
+    # print(params)
+    # listapi = os.environ['api_resource_uri']
+    # data = json.loads(listapi)
+    for i, d in enumerate(data):
+        flag = True
+        if d["authMethod"] == "basic":
+            del d["oauthurlPathkey"]
+        for k, v in d.items():
+            if d[k] in [None, '', ' ']:
+                logging.error(f"Please provide the value for '{k}' for '{d['servicename']}' service")
+                flag = False
+                break
+        if flag:
```

Comment:
`flag` should be renamed to a comprehensive name, some for the `i`, `d` and `k`, `v` variables to make it easier readable